src/runner.rs

```diff
@@ -25,6 +24,59 @@ pub fn runner(requirements: &[Requirements]) -> ExitCode {
     conclusion.exit_code()
 }
 
+fn find_tests(args: &Arguments, requirements: &[Requirements]) -> Vec<Trial> {
+    let tests: Vec<_> = if let Some(exact_filter) = exact_filter(args) {
+        let exact_tests: Vec<_> = requirements
+            .iter()
+            .flat_map(|req| req.exact(exact_filter))
+            .collect();
+
+        if is_nextest() {
+            if exact_tests.is_empty() {
+                panic!("Failed to find exact match for filter {exact_filter}");
+            } else if exact_tests.len() > 1 {
+                panic!(
+                    "Only expected one but found {} exact matches for filter {exact_filter}",
+                    exact_tests.len()
+                );
+            }
+        }
+        exact_tests
+    } else if is_full_scan_forbidden() {
+        panic!("Exact filter was expected to be used");
+    } else {
+        let mut tests: Vec<_> = requirements.iter().flat_map(|req| req.expand()).collect();
+        tests.sort_unstable_by(|a, b| a.name().cmp(b.name()));
+        tests
+    };
+    tests
+}
+
+fn is_nextest() -> bool {
+    std::env::var_os("NEXTEST")
+        .filter(|var| var == "1")
+        .is_some()
+}
+
+fn is_full_scan_forbidden() -> bool {
+    #[cfg(test)]
+    {
+        std::env::var_os("DATATEST_FULL_SCAN_FORBIDDEN")
```

Comment:
Could you prefix the env var with a double underscore to indicate that it's private? Don't want this to be part of the public API at the moment.

Reply:
I tried to make it private with the `#[cfg(test)]` condition, but I realise now that the datatest code is almost always going to be compiled under `#[cfg(test)]`.

Short of creating a custom feature for the crate, I don't think there's a way that I can block this environment variable from leaking into the public version. I guess the best we can do is the `__` prefix?

Reply:
Actually, I'm pretty sure tests using datatest-stable aren't going to have cfg(test) for datatest-stable. cfg(test) isn't transitive.

Reply:
Turns out that this version of the check wasn't working. After I changed it to use `std::env::var` the `run_example` test started failing during the initial `--list` run of the test harness.

I added a check in this function to exclude the `--list` case.

Reply:
Ahh yeah, good catch!

Reply:
> Actually, I'm pretty sure tests using datatest-stable aren't going to have cfg(test) for datatest-stable. cfg(test) isn't transitive.

Hmm maybe I shouldn't have removed it, then.